backend/main.py

```
@@ -155,14 +163,176 @@ async def root():
+        data_dir = Path("./data")
+        data_dir.mkdir(exist_ok=True)
+        test_file = data_dir / ".health_check"
+        test_file.write_text("ok")
+        test_file.unlink()  # Clean up test file
+        
+        health_status["services"]["filesystem"] = {
+            "status": "healthy",
+            "data_directory_writable": True
+        }
+    except Exception as e:
+        logger.error(f"Filesystem health check failed: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)